R/utils.R

```diff
@@ -46,43 +46,32 @@ get_teal_bs_theme <- function() {
 #' @keywords internal
 .include_parent_datanames <- function(datanames, join_keys) {
   ordered_datanames <- datanames
-  for (i in datanames) {
-    parents <- character(0)
-    while (length(i) > 0) {
-      parent_i <- teal.data::parent(join_keys, i)
-      parents <- c(parent_i, parents)
-      i <- parent_i
+  for (current in datanames) {
+    parents <- character(0L)
+    while (length(current) > 0 && !current %in% parents[-1]) {
```

Comment:
This looks like a bug. If dataset can have a single parent then neither ancestor can have two descendants if they are in a line. It means that either `&&` would throw because `length(current) > 1`  or condition will never be TRUE. Am I right?
 
```suggestion
    while (length(current) > 0) {
```

Reply:
The second part ensures it cannot go into infinite recursion. The `join_keys` data structure prevents that, but I think we should ensure that is never an option.

Reply:
Good point, no need to double check this here